.github/workflows/nightly.yml

```diff
@@ -103,93 +102,250 @@ jobs:
+        uses: actions/checkout@v1
+        with:
+          path: main
       - name: Extract branch name & commit
         id: extract_branch
         run: |
-          $env:git_hash = git.exe rev-parse --short=7 "$env:GITHUB_SHA"
-          $env:branch = echo $env:GITHUB_REF#refs/heads/ | %{$_ -replace '\/','.'}
-      - uses: actions/setup-python@v1
+          echo "::set-output name=git_hash::$(git rev-parse --short=7 "$GITHUB_SHA")"
+          echo "::set-output name=branch::$(echo ${GITHUB_REF#refs/heads/} | sed 's/\//\./g')"
+          echo "::set-output name=commits_ahead::$(git rev-list $(git describe --tags $(git rev-list --tags --max-count=1)).. --count)"
+          echo "::set-output name=last_tag::$(git describe --tags $(git rev-list --tags --max-count=1))"
+      - name: get version for veriblock-pop-cpp
+        run: echo "::set-env name=VERIBLOCK_POP_CPP_VERSION::$(awk -F '=' '/\$\(package\)_version/{print $NF}' $PWD/depends/packages/veriblock-pop-cpp.mk | head -n1)"
+      - name: veriblock-pop-cpp release version
+        run: echo Using veriblock-pop-cpp version $VERIBLOCK_POP_CPP_VERSION
+      - name: add bitcoin ppa
+        run: sudo add-apt-repository ppa:bitcoin/bitcoin
+      - name: install build requirements
+        run: sudo apt-get install build-essential libtool autotools-dev automake pkg-config bsdmainutils python3
+      - name: install dependencies
+        run: sudo apt-get install libevent-dev libboost-system-dev libboost-filesystem-dev libboost-test-dev libboost-thread-dev libminiupnpc-dev libzmq3-dev libdb4.8-dev libdb4.8++-dev libqt5gui5 libqt5core5a libqt5dbus5 qttools5-dev qttools5-dev-tools libqrencode-dev libubsan0 libubsan1 libtsan0 libasan4
+      - name: install sonar
+        env:
+          SONAR_CLI_VERSION: 4.2.0.1873
+        run: |
+          (
+          set -e; \
+          mkdir -p /opt/sonar; \
+          curl -L -o /tmp/sonar.zip https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-${SONAR_CLI_VERSION}-linux.zip; \
+          unzip -o -d /tmp/sonar-scanner /tmp/sonar.zip; \
+          mv /tmp/sonar-scanner/sonar-scanner-${SONAR_CLI_VERSION}-linux /opt/sonar/scanner; \
+          sudo ln -s -f /opt/sonar/scanner/bin/sonar-scanner /usr/local/bin/sonar-scanner; \
+          rm -rf /tmp/sonar*
+          )
+      - name: install veriblock-pop-cpp
+        run: |
+          (
+          cd /opt;
+          wget https://github.com/VeriBlock/alt-integration-cpp/archive/${VERIBLOCK_POP_CPP_VERSION}.tar.gz;
+          tar -xf ${VERIBLOCK_POP_CPP_VERSION}.tar.gz;
+          cd alt-integration-cpp-${VERIBLOCK_POP_CPP_VERSION};
+          mkdir build;
+          cd build;
+          cmake .. -DCMAKE_BUILD_TYPE=Release -DTESTING=OFF;
+          make -j2;
+          sudo make install
+          )
+      - name: autogen
+        run: ./autogen.sh
+      - name: configure
+        run: ./configure
+          --disable-asm
+          --with-sanitizers=address,undefined
```

Comment:
can we change this build to be matrix build? same as we do in alt-cpp: https://github.com/VeriBlock/alt-integration-cpp/blob/master/.github/workflows/ci.yml#L19

would be 
```
matrix:
  flags: [
    "address,undefined",
    "thread"
  ]
```

Reply:
For now, I would prefer to leave it as-is, to make sure that it runs and dependencies are met.

I will create a test branch to implement the matrix and we can re-factor the CI and clean it up a bit when we add the Windows build back in.